src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqBinding.cs

```diff
@@ -0,0 +1,156 @@
+            : this(Protocols.DefaultProtocol)
+        { }
+
+        /// <summary>
+        /// Uses the broker specified by the given hostname and port with
+        /// Protocols.DefaultProtocol.
+        /// </summary>
+        /// <param name="hostname">The hostname of the broker to connect to</param>
+        /// <param name="port">The port of the broker to connect to</param>
+        public RabbitMqBinding(string hostname, int port)
+            : this(hostname, port, Protocols.DefaultProtocol)
+        { }
+
+        /// <summary>
+        /// Uses the broker and protocol specified
+        /// </summary>
+        /// <param name="hostname">The hostname of the broker to connect to</param>
+        /// <param name="port">The port of the broker to connect to</param>
+        /// <param name="protocol">The protocol version to use</param>
+        public RabbitMqBinding(string hostname, int port, IProtocol protocol)
+            : this(protocol)
+        {
+            HostName = hostname;
+            Port = port;
+        }
+
+        /// <summary>
+        /// Uses the broker, login and protocol specified
+        /// </summary>
+        /// <param name="hostname">The hostname of the broker to connect to</param>
+        /// <param name="port">The port of the broker to connect to</param>
+        /// <param name="username">The broker username to connect with</param>
+        /// <param name="password">The broker password to connect with</param>
+        /// <param name="virtualHost">The broker virtual host</param>
+        /// <param name="maxMessageSize">The largest allowable encoded message size</param>
+        /// <param name="protocol">The protocol version to use</param>
+        public RabbitMqBinding(string hostname, int port,
+                               string username, string password, string virtualHost,
+                               long maxMessageSize, IProtocol protocol)
+            : this(protocol)
+        {
+            HostName = hostname;
+            Port = port;
+            Transport.UsernameConfigKey = username;
+            Transport.PasswordConfigKey = password;
+            Transport.VirtualHost = virtualHost;
+            MaxMessageSize = maxMessageSize;
+
+        }
+
+        /// <summary>
+        /// Uses the specified protocol. The broker must be set before use.
+        /// </summary>
+        /// <param name="protocol">The protocol version to use</param>
+        public RabbitMqBinding(IProtocol protocol)
+        {
+            BrokerProtocol = protocol;
+            Name = "RabbitMQBinding";
+            Namespace = "http://schemas.rabbitmq.com/2007/RabbitMQ/";
+
+            Initialize();
+        }
+
+        public override BindingElementCollection CreateBindingElements()
+        {
+            //Transport.HostName = HostName;
+            //Transport.Port = Port;
+            Transport.BrokerProtocol = BrokerProtocol;
+            if (MaxMessageSize != DefaultMaxMessageSize)
+            {
```

Comment:
It's better to have the MaxMessageSize property be a passthrough to setting Transport.MaxReceivedMessageSize rather than needing to synchronize them.